.github/workflows/build.yml

```
@@ -0,0 +1,207 @@
+name: Test & Build
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    name: Test User Management
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+      - uses: unfor19/install-aws-cli-action@v1
+      - name: Install modules
+        run: go mod download
+      - name: Setup gotestsum
+        uses: autero1/action-gotestsum@v1.0.0
+        with:
+          gotestsum_version: 1.6.4
+      - run: mkdir -p /tmp/test-results
+      - name: Download pact tools
+        run: |
+          curl -LO https://github.com/pact-foundation/pact-ruby-standalone/releases/download/v1.88.6/pact-1.88.6-linux-x86_64.tar.gz
+          tar xzf pact-1.88.6-linux-x86_64.tar.gz
+          echo "$PWD/pact/bin" >> $GITHUB_PATH
+      
+      - name: Run Tests
+        run: |
+          gotestsum --junitfile /tmp/test-results/unit-tests.xml -- ./... -coverprofile=/tmp/test-coverage.txt
+
+      - name: Publish pacts
+        env:
+          PACT_DIR: ./pacts
+          PACT_BROKER_URL: https://pact-broker.api.opg.service.justice.gov.uk
+          PACT_BROKER_USERNAME: admin
+          PACT_BROKER_PASSWORD: ${{ secrets.PACT_BROKER_PASSWORD }}
+        run: PACT_TAG=${{ github.head_ref }} PACT_CONSUMER_VERSION=${{ github.sha	}} go run internal/pact/publish.go
```

Just noting this whilst it's on my mind:
```suggestion
        run: PACT_TAG=${{ github.head_ref }} PACT_CONSUMER_VERSION=${{ github.event.pull_request.head.sha }} go run internal/pact/publish.go
```